lib/HeroBonus.cpp

```diff
@@ -850,6 +850,14 @@ void CBonusSystemNode::removeBonus(const std::shared_ptr<Bonus>& b)
 	CBonusSystemNode::treeHasChanged();
 }
 
+void CBonusSystemNode::removeBonuses(const CSelector &selector)

```

Comment:
And one more request. We already had ```CBonusSystemNode::popBonuses``` that doing almost (?) the same thing except it's also pop bonuses from children. I suppose secondary skills not affect children nodes so it's should be okay.

Though I think we need to rename ```popBonuses``` now to make it easier to grasp where you should use it or ```removeBonuses```. At least it's seems like something important for me.

Reply:
Hadn't spotted that function. So I fully agree, popBonuses is not ideal, especially since we have a method removeBonus for single bonus. My suggestion would be `removeBonusesRecursive` or similar.

Reply:
Makes sense to me. Go for ```removeBonusesRecursive``` then.